CRM/Pledge/Form/Pledge.php

```diff
@@ -156,10 +156,7 @@ public function setDefaultValues(): array {
     );
 
     // assign status.
-    $this->assign('status', CRM_Utils_Array::value(CRM_Utils_Array::value('status_id', $this->_values),
-      $pledgeStatus,
-      $defaultPledgeStatus
-    ));
+    $this->assign('status', $pledgeStatus[$this->_values['status_id']] ?? $defaultPledgeStatus);
```

Comment:
@colemanw the test failure seems like related: 
```
CRM_Pledge_Form_PledgeTest::testPostProcess
Undefined array key "status_id"

/home/homer/buildkit/build/build-2/web/sites/all/modules/civicrm/CRM/Pledge/Form/Pledge.php:159
```

Reply:
Thanks @monishdeb. I think this fixes it.

Reply:
Yep looks good!